### DISCUSSION TITLE
No Insteon messages received from KPL scene buttons

### DISCUSSION THREAD

**Discussion by kpfleming** [TD22057/insteon-mqtt#440] [Q&A]
This is probably somehow related to #18, but anyway...

I've just installed a 6-button KPL and FL. I used the entirely manual setup process to link all four scene buttons to the FL's fan speed controls, and also put them into radio button mode (I don't have any Hubs and don't use the Insteon app, so manual mode is all I have!).

In the logs from insteon-mqtt I see Insteon messages from the KPL ON and OFF buttons (and the appropriate MQTT messages are sent as well), but I don't see any Insteon messages at all when the scene buttons are pressed.

**Comment by krkeegan** [comment]
Hmm, no idea what would happen in radio mode.  The only things I can suggest are to run `refresh --force` to ensure we have a proper understanding of the link table, and `pair` to make sure that the individual buttons are informing the PLM of their changes.  That may help, it may not.

**Reply by RememberTilman** [reply]
what is the pair command here? pair aa.bb.cc, where aa..bb.cc is the KPL? I’ve done that, plus the scene.yaml definitions, and LINKING commands (to and from the modem/KPL), to no avail. For the life of me, I can’t get a message from the KPL. The scene.yaml seems to work OK, because pushing the button causes the device to respond. JOIN/PAIR seems to work, because I can command a button LED to turn on. But pushing a button doesn’t send a message.

**Reply by house97** [reply]
Is the PLM part of the scene? I may be mistaken, but I think that if the KPL button is a controller, and it's controlling the FL, and the PLM is not part of the scene, then neither the KPL or FL will send any messages to the PLM.

**Comment by kpfleming** [comment]
That's an interesting question; it's not necessary to make the PLM part of scenes with other controllers, but maybe it is with the KPL.

**Comment by kpfleming** [comment]
Woot! The changes made in #453 have fixed this issue. I now see Insteon messages arriving from the modem when I press scene buttons on the KPL. MQTT messages are emitted showing the changes in state for the linked FanLinc too.
